Create mouse-reconcile-desktop.md

kikisaints opened the pull request:
Initial draft of reconciliation doc.
**Please note a few things:** 

- @harinikmsft and @chrisglein this is in a fork of the main because it is still a draft of this reconciliation doc. 
It has been a bit of a battle to try and grok the web implementation, our own tech, and the delta needed between them in a way that make sense in our system, so I ask that you please review and provide feedback that I am calling out the right items here
- Not all issues are created yet, but are stubbed out. **They will be created before official main merge** but I want to ensure I've identified the right issues before I start creating them will-nilly in our main repo
- I'm most unsure about the necessity/accuracy of the onPointer callbacks and declaritive properties, if you two could lend some feedback (or '@' the right people) that would help a lot.

Thanks!

kikisaints committed:
Create mouse-reconcile-desktop.md

Initial draft of reconciliation doc.

kikisaints committed:
added anothing issue needed to complete mouse reconciliation

harinikmsft commented on vnext/proposals/active/mouse-reconcile-desktop.md:
```diff
@@ -0,0 +1,91 @@
+- Ensuring mouse behavior is consistent across the two platforms we support
+- Once our two desktop platforms are aligned with React Native Web, we upstream this implementation to core [issue TBD]()
+
+## `hovered`
```
Is this prop only for `Pressable` or should this be introduced in `View` or others as well for completeness? What does web do? Perhaps @ necolas can help you figure this out.

harinikmsft commented on vnext/proposals/active/mouse-reconcile-desktop.md:
```diff
@@ -0,0 +1,91 @@
+
+Where `IHandledPointerEvents` is a new type which takes the following parameters:
+- A constrained string parameter named `pointerState` to declare the pointer state (Over, Hover, Out) that is of interest to the JS layer
+- An `eventPhase` parameter of type `EventPhase` to declare the routing phase of interest to the JS layer.
```
Is this `EventPhase` just the generic one that is also being used in KeyboardEvents? Meaning, the parameters indicate bubbling/tunneling like the web, correct? Maybe good to add links to the React docs for these to make it simpler for implementation coherence.

harinikmsft commented on vnext/proposals/active/mouse-reconcile-desktop.md:
```diff
@@ -0,0 +1,91 @@
+- The hover events/indication for overlapping content prioritizes items "closer" in z-order [issue #5901](https://github.com/microsoft/react-native-windows/issues/5901)
+- Remove the tapped event handler on the `View` component [issue #2349](https://github.com/microsoft/react-native-windows/issues/2349)
+
+## onPointerXX callbacks
```
There were a lot of other `onPointerXX` callbacks in the proposal discussion, is this now a smaller constrained list for unblocking immediate scenarios? If yes, that may be good to capture along with a short passage on other possible callbacks that can be revisited later -capturing this for posterity will be useful.

harinikmsft commented:
Looks good @kikisaints . I don't think you need to stage a review separately for this, you should start a PR to the main docs site directly and add necolas, acoates-ms, tom-un, kmelmon, falselobster for feedback and iterate in the open. Looks great, go for it!

kikisaints committed:
added callbacks table of events not currently being considered

kikisaints committed:
Updates to sections, wording, events being reconciled

Added up and down pointer events. Fixed some sections and moved them down. Updated examples to better (hopefully) reflect usages.
